deploy.go

```diff
@@ -257,22 +258,21 @@ func newDeploymentRequest(c *cli.Context, ref string, env string) (*github.Deplo
 		}
 	}
 
-	var contexts *[]string
-	if c.Bool("force") {
-		s := []string{}
-		contexts = &s
-	}
+  // Build the payload, which goes...?
+  payload, err := json.Marshal(map[string]interface{}{
+    "force": c.Bool("force"),
+  })
+  if err != nil {
+    return nil, err
+  }
 
 	return &github.DeploymentRequest{
 		Ref:              github.String(ref),
 		Task:             github.String("deploy"),
 		AutoMerge:        github.Bool(false),
 		Environment:      github.String(env),
-		RequiredContexts: contexts,
```

Comment:
Not passing `RequiredContexts` definitely breaks the force flag. I think `Payload` is just metadata passed to whatever is handling deployments, so we can ignore that. By default (if `RequiredContexts` is nil), GitHub will require all contexts/checks to be successful. It seems to be the force flag disables that by passing an empty list rather than nil, which tells the GitHub API that no contexts should be required.

Reply:
Yeah, the problem is that in go, an empty slice will be dropped from the JSON when omitempty is set, which it is for DeploymentRequest. This is an api change in the github sdk that (i think) just breaks the way this code is trying to use force.

Reply:
Oh, WTF. So it's possible this never worked correctly?

Reply:
Apparently `omitempty` _does_ still include the key if the value is a pointer to an empty array:
https://play.golang.org/p/pwuCgZaSuUu

Reply:
Yes, sorry if that wasn't clear. That's why it used to work, but with github changing their api like this, it will no longer work, afaict. I'm honestly not sure how to disable checks with this new api, but I haven't dug into it. I really just want to get this all working on go modules and fix the auth issues so it's usable again. I think it's a good fallback to have working if slashdeploy fails.

Reply:
So this is fixed in https://github.com/google/go-github/commit/b2f8d2ec4070d3535b493308749e1c1f5c6c8e55, which we already had in the old vendor: https://github.com/remind101/deploy/blob/4f9b60bed8351ef9bc9aeefa0e2435d19e9e0a55/vendor/github.com/google/go-github/github/repos_deployments.go#L33

So how come this change is using `github.com/google/go-github v0.0.0-20150126182318-7ea4ee6d2226` in `go.mod`? That version is _older_ than what we had in vendor, I think.

Reply:
It looks like https://github.com/remind101/deploy/blob/4f9b60bed8351ef9bc9aeefa0e2435d19e9e0a55/vendor/vendor.json#L94-L97 is wrong, because https://github.com/google/go-github/blob/7ea4ee6d222607c11ea86e99a6f6723beeae785d/github/repos_deployments.go#L33 doesn't match https://github.com/remind101/deploy/blob/4f9b60bed8351ef9bc9aeefa0e2435d19e9e0a55/vendor/github.com/google/go-github/github/repos_deployments.go#L33

Reply:
I think it's worth trying to update go-github at least, before this is merged.